test/test_init.py

```diff
@@ -0,0 +1,90 @@
+from opensquirrel import Circuit, CircuitBuilder
+from opensquirrel.ir import Bit
+
+
+def test_init() -> None:
+    qc = Circuit.from_string(
+        """
+        version 3.0
+
+        qubit[2] q
+        bit[2] b
+        H q[0]
+        CNOT q[0], q[1]
+
+        init q[0]
+        b = measure q
+        """,
+    )
+    assert (
+        str(qc)
+        == """version 3.0
+
+qubit[2] q
+bit[2] b
+
+H q[0]
+CNOT q[0], q[1]
+init q[0]
+b[0] = measure q[0]
+b[1] = measure q[1]
+"""
+    )
+
+
+def test_init_sgmq() -> None:
+    qc = Circuit.from_string(
+        """
+        version 3.0
+
+        qubit[4] q
+
+        H q[0]
+        H q[1:2]
+        init q[2:3]
+        H q[3]
+        init q[0:1]
+        """,
```

Comment:
Same here.
If we want to test `init` in the middle of a circuit, I would put it in a separate test.
BTW, I would propose to write this set of tests as a parametrized test. You can write the programs and the outputs in a single line. Guy wrote recently one like that (`test_simplest` in test_libqasm.py). E.g.:

```
    (
        "version 3.0; qubit[4] q; init q[2:3]",
        "version 3.0\n\nqubit[4] q\n\ninit q[2]\ninitq[3]",
    )
```

```suggestion
        version 3.0

        qubit[4] q

        init q[2:3]
        """,
```